### ISSUE TITLE
Support driver models with read paths (e.g. Dropbox or Google Drive)

### ISSUE THREAD

**Issue by kantai** [blockstack/gaia#121]
It would be nice to be able to support some drivers with more complex requirements (i.e., because the storage provider doesn't provide the ability to pre-determine the URL for a file).

These drivers would require that the gaia hub be invoked on the _read path_ -- so if we want to support something like this, we need a "read-path" gaia driver type. For this spec, I'd propose a `get` endpoint:

```
GET /read/
```

Which will return the data (or, in an ideal world, a `301`, but I think that these kinds of backends wouldn't typically have CORS headers set).

**Comment by markmhx** [comment]
@kantai mind providing a quick update here on the work (ongoing or otherwise), and perhaps the proposed driver model and API?

Also, I've added this to the "Expand storage driver support" project but see @jcnelson also added it earlier to "User-selectable Gaia Storage". My sense is that it's not exactly required to support user-selectable storage, is it?

**Comment by jcnelson** [comment]
This is more closely aligned with "Expand storage driver support."

**Comment by kantai** [comment]
Yep -- this is more in the "expand storage driver support", which may ultimately play a part in making running a gaia hub more user-friendly.
